test/arrayfire_test.cpp

```diff
@@ -1390,6 +1391,116 @@ INSTANTIATE(long long);
 INSTANTIATE(unsigned long long);
 #undef INSTANTIATE
 
+template<typename T>
+struct sparseCooValue {
+    int row = 0;
+    int col = 0;
+    T value = 0;
+    sparseCooValue(int r, int c, T v) : row(r), col(c), value(v) {}
+};
+
+template<typename T>
+void swap(sparseCooValue<T> &lhs, sparseCooValue<T> &rhs) {
+    std::swap(lhs.row, rhs.row);
+    std::swap(lhs.col, rhs.col);
+    std::swap(lhs.value, rhs.value);
+}
+
+template<typename T>
+bool operator<(const sparseCooValue<T> &lhs, const sparseCooValue<T> &rhs) {
+    if (lhs.row < rhs.row) {
+        return true;
+    } else if (lhs.row == rhs.row && lhs.col < rhs.col) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+template<typename T>
+std::ostream &operator<<(std::ostream &os, const sparseCooValue<T> &val) {
+    os << "(" << val.row << ", " << val.col << "): " << val.value;
+    return os;
+}
+
+template<typename T>
+bool isZero(const sparseCooValue<T> &val) {
+    return val.value == 0.;
```

Comment:
For future reference when updating tests for sparse solve, improve comparison w/float to within threshold